### PR TITLE
pkg/identity: Move GetCIDKeyFromK8sLabels to GlobalIdentity

### PR DESCRIPTION
**Pull request by ovidiutirla**
Moves the method from `ciliumidentity` package to GlobalIdentity and makes the method more generic by accepting the source to be used in mapping.

Related #27752

### CODE REVIEW

File: pkg/identity/key/global_identity.go [hunk 1]
```diff
@@ -70,3 +72,9 @@ func (gi *GlobalIdentity) PutValue(key, value any) allocator.AllocatorKey {
 func (gi *GlobalIdentity) Value(key any) any {
 	return gi.metadata[key]
 }
+
+func GetCIDKeyFromLabels(allLabels map[string]string, source string) *GlobalIdentity {
```
Comment (derailed) [inline pkg/identity/key/global_identity.go:76]: Perhaps we should add some tests on this convenience call?

**Review by derailed — APPROVED** [review]
@ovidiutirla Nice work!

File: pkg/identity/key/global_identity.go (hunk 1 above)
Comment (ovidiutirla) [reply pkg/identity/key/global_identity.go:76]: hey @derailed, thank you for your review. I've added a few tests.

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

**Review by pippolo84 — APPROVED** [review]
Thanks! :rocket:

**Pull request merged**